onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -5336,17 +5336,52 @@ def aten_slice_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::slice_scatter", trace_only=True)
 def aten_slice_scatter(
-    self: TensorType,
-    src: TensorType,
-    dim: int = 0,
-    start: Optional[INT64] = None,
-    end: Optional[INT64] = None,
-    step: INT64 = 1,
-) -> TensorType:
+    self: TTensor,
+    src: TTensor,
+    dim: int,
+    start: int,
+    end: int,
+    step: int = 1,
```

Comment:
```suggestion
    dim: int=0,
    start: Optional[INT64]=None,
    end: Optional[INT64]=None,
    step: INT64 = 1,
```

start etc can be symint (dynamic) it looks like. Do we also want to consider None as inputs?

Reply:
Althouth 'start' and 'end' can be None in signature, but actually 'start' must be specified
Assert(start is not None)
And, 'end' also must be specified, and end-start must be equal to the size of 'src'
Assert(end-start == shape(src) > 0)
Try torch sample to get more information:
https://pytorch.org/docs/master/generated/torch.slice_scatter.html?highlight=slice_scatter#torch.slice_scatter

Reply:
Why do we need them to be int?

Reply:
From signature, they can be INT64, both INT64 and int can works well.
From fact intention, they are all represent to scalar(index).
Even we set dim: Optional[INT64] in the public function, we still need to set dim: int in the private function, because op.ScatterElements(... axis=dim) still need dim as int type.

Reply:
ok dim needs to be int. Can we set start and end to be INT64 since they are SymInts in the signature? In PyTorch SymInts are dynamic.

Reply:
OK.